exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -177,6 +182,18 @@ public void run() {
     });
   }
 
+  /** If we are in distributed mode, disallow using loopback address to register new drillbit
+   *  unless zookeeper is on the same node with drillbit. */
+  private void checkLoopbackAddress(String address) throws DrillbitStartupException, UnknownHostException {
+    if (isDistributedMode) {
+      String coordAddress = StringUtils.substringBefore(config.getString(ExecConstants.ZK_CONNECTION), ":");
+      if (!InetAddress.getByName(coordAddress).isLoopbackAddress()
```

Comment:
In distributed mode, no bit should be allowed to register with a loopback address.

If bit A is allowed to register with a loopback address and bit B tries to communicate with bit A, bit B would be sending messages to itself.